packages/react-provider/src/ws.ts

```diff
@@ -1,74 +1,91 @@
-import { ICourierEventCallback } from "./transports/types";
+import { ICourierEventCallback, ICourierMessage } from "./transports/types";
+import ReconnectingWebSocket from "reconnecting-websocket";
 
 export class WS {
-  connection?: WebSocket;
+  connection?: ReconnectingWebSocket;
+  private subscriptions: Array<{
+    channel: string;
+    event?: string;
+    callback: ICourierEventCallback;
+  }>;
   protected connected;
   protected messageCallback;
   private url: string;
+  private clientKey: string;
 
-  constructor({ url }) {
+  constructor({ url, clientKey }: { url: string; clientKey: string }) {
     this.messageCallback = null;
     this.connection = undefined;
     this.connected = false;
     this.url = url;
+    this.clientKey = clientKey;
+    this.subscriptions = [];
   }
 
-  connect(clientKey: string): void {
-    const url = `${this.url}/?clientKey=${clientKey}`;
-    this.connection = new WebSocket(url);
-    if (!this.connection) {
-      console.error("error creating websocket connection");
-      return;
-    }
-    this.connection.onopen = () => {
-      this.connected = true;
-    };
-    this.connection.onclose = () => {
-      // i want to watch and see if we get connection closed events
-      // i think we will want to reconnect when this happens
-      console.warn("ws connection closed");
-    };
+  connect(): void {
+    this.connection = new ReconnectingWebSocket(
+      `${this.url}/?clientKey=${this.clientKey}`
+    );
+
+    this.connection.onopen = this.onOpen.bind(this);
     this.connection.onmessage = this.onMessage.bind(this);
   }
 
+  onOpen(): void {
+    this.connected = true;
```

Comment:
should we also do `this.connected = false` in `close()`?

Reply:
yeah prob, good idea